opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -189,6 +231,36 @@ public function index() {
         $this->response->setOutput($this->render());
     }
 
+    protected function handleLayoutSaving() {
+        if (!$this->user->hasPermission('modify', 'module/smaily_for_opencart')) {
+            return;
+        }
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Check if layout settings are in POST & that it's an array.
+        if (!empty($this->request->post['smaily_for_opencart_module']) && is_array($this->request->post['smaily_for_opencart_module'])) {
+            $smaily_modules_san = [];
+            // Declare layout module settings in POST.
+            $smaily_modules = $this->request->post['smaily_for_opencart_module'];
+            // Loop over each array and validate array keys.
+            foreach ($smaily_modules as $module) {
+                // Check if layout_id is number, if not default to 0.
+                $module['layout_id'] = (isset($module['layout_id']) && is_numeric($module['layout_id'])) ? $module['layout_id'] : '0';
+                // Check if position is one out of 4 available, if not default to content_bottom
+                $module['position'] = (isset($module['position']) && in_array(
+                    $module['position'], array('content_top', 'content_bottom', 'column_left', 'column_right')
+                )) ? $module['position'] : 'content_bottom';
+                // Check if status is 1 or 0, if none default to 0
+                $module['status'] = (isset($module['status']) && $module['status'] == '1' || '0') ? $module['status'] : '0';
```

Comment:
For safety and readability I would add parenthesis around `'1' || '0'`.

Reply:
I am not sure if this even works though. I know you can write it like this in Javascript, but I am not sure it works in PHP.